### ISSUE TITLE
Compilation fails when enumerated type name is 'b' or 'x'

### ISSUE THREAD

**Issue by CervEdin** [MiniZinc/libminizinc#726]
Declaring enums with either type name `b` or `x` fails to compile

```
enum b = { e1 }; % Error: type-inst must be par set but is 'bool'
enum x = { e2 }; % Error: type-inst cannot be array
```

Running MiniZinc version 2.6.4, build 570825159

Also reproduced in
```
2.4.2
2.5.5
2.7.4
```

**Comment by Dekker1** [comment]
I would imagine that some of the code introduced for enumerated types uses `x` and `b`. I'll have a look if I can find them and replace them with proper introduced variables.

**Comment by cyderize** [comment]
This should now be fixed in 2.8.0
